### ISSUE TITLE
Configs: extend `@typescript-eslint/eslint-recommended` with additional redundant rules

### ISSUE THREAD

**Issue by maxpatiiuk** [typescript-eslint/typescript-eslint#8155]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22preset+config+change%22+) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Description

The `@typescript-eslint/eslint-recommended` config is great as it disables many of the redundant rules for projects using TypeScript. However, there are a few more redundant ESLint rules that should be disabled. Could those be added to `@typescript-eslint/eslint-recommended`? In particular:

- [valid-typeof](https://eslint.org/docs/latest/rules/valid-typeof)
- [no-unused-labels](https://eslint.org/docs/latest/rules/valid-typeof)
- [no-unused-vars](https://eslint.org/docs/latest/rules/no-unused-vars)
- [use-isnan](https://eslint.org/docs/latest/rules/use-isnan)

   though, seems like typescript does not report `foo === Number.NaN` - it probably should, so maybe rule should stay enabled for now:
   ![](https://github.com/typescript-eslint/typescript-eslint/assets/40512816/04e2674c-2458-4d8d-bbcb-58a59fe3e16c)

Also, while [no-unsafe-negation](https://eslint.org/docs/latest/rules/valid-typeof) is redundant with TypeScript, unlike TypeScript, it includes an auto-fix, thus it may be desirable to not disable it in `@typescript-eslint/eslint-recommended`

### Impacted Configurations

`@typescript-eslint/eslint-recommended`

**Comment by maxpatiiuk** [comment]
Hi helpful bot. The image is preceded by a description, so the image having empty alt text is valid

**Comment by auvred** [comment]
> no-unused-labels

TSConfig has [`allowUnusedLabels`](https://www.typescriptlang.org/tsconfig#allowUnusedLabels) option, I think it's up to user to decide whether to disable `no-unused-labels` based on tsconfig or not

> no-unused-vars

Same as above. TSConfig has [`noUnusedLocals`](https://www.typescriptlang.org/tsconfig#noUnusedLocals)

It may happen that a user does not have the options mentioned above in tsconfig, and we turn off eslint rules in our config. 
And the user is left with an uncaught error

**Comment by JoshuaKGoldberg** [comment]
> Hi helpful bot. The image is preceded by a description, so the image having empty alt text is valid. Providing redundant alt-text is only going to reduce accessibility

@maxpatiiuk Not quite. Please use proper alt text to fill out image text.
* General case: programmatically linking the image to alt text means screenreaders can announce the text in line with the image.
* This specific case: the image has a lot of content -code, squiggly lines, the editor tooltip- that aren't indicated by the comment. So even if the rest of the comment was linked to the image, it wouldn't be sufficient.
That being said, it is irksome that the bot is posting twice. https://github.com/github/accessibility-alt-text-bot/issues/31 😕

**Comment by bradzacher** [comment]
> valid-typeof

This is intentionally not included. See https://github.com/typescript-eslint/typescript-eslint/issues/3797

Because this is valid TS:
```ts
typeof foo === undefined
```

And is the most common typo bug with `typeof`

> no-unused-vars
> no-unused-labels

As auvred mentions - the user may turn off the TS checks that mimic these rules. So we wont turn them off in a recommended set.

Additionally it's best if the "linting" features are left to a linter - our recommendation is to turn these off and use the lint rules instead. The TS team actually laments about having added linting flags to TS.

> use-isnan

As you mentioned - there is at least one case TS doesn't catch - so we can't turn it off.

> no-unsafe-negation

An autofix suggestions are a very useful thing, for sure. I think that we should keep it on for that reason alone, tbh.

---

Given the above rejections of each suggested rule. We shall close this as a won't fix. Thanks for filing!

**Comment by maxpatiiuk** [comment]
Thank for the details, and I agree with your decisions, except:

> An autofix suggestions are a very useful thing, for sure. I think that we should keep it on for that reason alone, tbh.

But the rule is currently being disabled by your default config:

https://github.com/typescript-eslint/typescript-eslint/blob/675e987ca1d13244c03d7e09d4e42c6539689d9a/packages/eslint-plugin/src/configs/eslint-recommended.ts#L26C10-L26C19

Thus, consider not disabling it
